src/Symfony/Component/Messenger/Event/AbstractWorkerMessageEvent.php

```diff
@@ -29,6 +29,11 @@ public function getEnvelope(): Envelope
         return $this->envelope;
     }
 
+    public function setEnvelope(Envelope $envelope)
+    {
+        $this->envelope = $envelope;
```

Comment:
making the envelope mutable means, all the dispatching of events would need to be changed to refetch the potentially modified envelope. -1 on doing this.

Reply:
Yes, this part is missing. But do you see it as a huge problem to (effectively) add `$envelope = $event->getEnvelope()`? I made the events immutable originally because I feel that it's always better to be conservative *until* we have a use-case. But this PR now does have a use-case.

Reply:
sorry i missed that...
in case this PR is accepted, i've fixed that and refetched envelope from mutable events.
i've added two new tests which check that events mutability is handled

Reply:
Did we need to add this in the abstract class? This will affect (on master currently) 3 classes. Maybe we need hook points in all of them? But in this PR, we only need a hook point in one of them.

Reply:
ok, i'll do that, it will be less error prone this way

Reply:
done :)